### PR TITLE
utils_conn: update configuration for TCP/TLS/UNIX connection

### PR DESCRIPTION
**Pull request by Yingshun**
Feature changes from libvirt 5.6, so update accordingly.
1. tcp_port changes to 'ListenStream' in
    /usr/lib/systemd/system/libvirtd-tcp.socket
2. tls_port changes to 'ListenStream' in
    /usr/lib/systemd/system/libvirtd-tls.socket
3. unix_sock_dir configuration changes to 'ListenStream' in
    /usr/lib/systemd/system/libvirtd.socket
4. unix_sock_rw_perms changes to 'SocketMode' in
    /usr/lib/systemd/system/libvirtd.socket

Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result on libvirt 5.6:**
```
(.libvirt-ci-venv-ci-runtest-Dyxj0x) [root@dell-per740-07 ~]# avocado run --vt-type libvirt --vt-machine-type q35 virsh.remote_with_unix.positive_testing.customized_unix_sock_dir virsh.remote_with_tcp.positive_testing.customized_ipv4_tcp_port remote_with_tls.positive_testing.customized_tls_port --vt-connect-uri qemu:///system
JOB ID     : 01d4372fd215915f2a447704189810338120b1c4
JOB LOG    : /root/avocado/job-results/job-2019-10-15T03.40-01d4372/job.log
 (1/3) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.customized_unix_sock_dir: /PASS (111.26 s)
 (2/3) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.positive_testing.customized_ipv4_tcp_port: PASS (79.79 s)
 (3/3) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.customized_tls_port: PASS (149.28 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 344.25 s
(.libvirt-ci-venv-ci-runtest-Dyxj0x) [root@dell-per740-07 ~]# libvirtd -V
libvirtd (libvirt) 5.6.0
```

**Test result on libvirt 4.5.0:**
```
 (1/3) type_specific.io-github-autotest-libvirt.virsh.remote_with_unix.positive_testing.customized_unix_sock_dir: -PASS (111.21 s)
 (2/3) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.positive_testing.customized_ipv4_tcp_port: PASS (64.92 s)
 (3/3) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.customized_tls_port: PASS (136.84 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 321.17 s
```

File: virttest/utils_conn.py [hunk 1]
```diff
@@ -1230,8 +1269,12 @@ def server_setup(self, on_local=False):
 
         # edit the /etc/libvirt/libvirtd.conf to add tls_port=$tls_port
         if tls_port != '16514':
-            pattern_to_repl = {r".*tls_port\s*=\s*.*": 'tls_port="%s"' % tls_port}
-            operate_libvirtdconf.sub_else_add(pattern_to_repl)
```
Comment (chunfuwen) [inline virttest/utils_conn.py]: maybe can be simplified as:
pattern_ro_repl=xx
if libvirt_version(5,6):
     pattern_ro_reply=yy
operate_libvirtdconf_sun_lese_add(pattern_ro_reply)

File: virttest/utils_conn.py [hunk 2]
```diff
@@ -1662,9 +1726,14 @@ def conn_setup(self):
 
         # edit the /etc/libvirt/libvirtd.conf to add unix_sock_dir arg
         if unix_sock_dir != '/var/run/libvirt':
-            pattern_to_repl = {r".*unix_sock_dir\s*=.*":
-                               'unix_sock_dir="%s"' % unix_sock_dir}
-            self.client_libvirtdconf.sub_else_add(pattern_to_repl)
+            if self.libvirt_ver:
+                pattern_to_repl = {r".*ListenStream\s*=.*":
+                                   'ListenStream=%s/libvirt-sock' % unix_sock_dir}
+                self.client_libvirtd_socket.sub_else_add(pattern_to_repl)
```
Comment (chunfuwen) [inline virttest/utils_conn.py]: see previous comment

File: virttest/utils_conn.py [hunk 3]
```diff
@@ -1724,7 +1798,13 @@ def conn_setup(self):
             try:
                 libvirtd_service = utils_libvirtd.Libvirtd(
                     session=client_session)
-                libvirtd_service.restart()
+                if self.libvirt_ver:
+                    libvirtd_service.stop()
+                    process.run("systemctl daemon-reload")
```
Comment (chunfuwen) [inline virttest/utils_conn.py]: we may need check the result for process.run().

Comment (Yingshun) [reply virttest/utils_conn.py]: will raise exception if cmd failed

File: virttest/utils_conn.py (hunk 1 above)
Comment (Yingshun) [reply virttest/utils_conn.py]: need to update different pattern in different configuration file here.

**Pull request merged**